crates/uv-pypi-types/src/conflicting_groups.rs

```diff
@@ -0,0 +1,192 @@
+use uv_normalize::{ExtraName, PackageName};
+
+/// A list of conflicting groups pre-defined by an end user.
+///
+/// This is useful to force the resolver to fork according to extras that have
+/// unavoidable conflicts with each other. (The alternative is that resolution
+/// will fail.)
+#[derive(
+    Debug, Default, Clone, Eq, PartialEq, serde::Deserialize, serde::Serialize, schemars::JsonSchema,
+)]
+pub struct ConflictingGroupList(Vec<ConflictingGroups>);
```

Comment:
I am wondering if the terminology here should just around something more general, like "Conflicts", since it has to cover both extras and dependency groups. The use of "groups" here makes it somewhat overloaded.

Reply:
Yeah @zanieb suggested just using `conflicts`, so I think I'll switch to that. I don't like that it doesn't say _what_ is conflicting, but "extra" and "group" are already such general names that it's hard to think of something more general than that but still captures their essence. And in theory, we could add more types of conflicts in the future, although the use cases aren't quite clear.